near-contract-standards/src/fungible_token/implementation.rs

```diff
@@ -0,0 +1,256 @@
+        receiver_id: ValidAccountId,
+        amount: U128,
+    ) -> U128 {
+        let amount: Balance = amount.into();
+        let sender_id: AccountId = sender_id.into();
+        let receiver_id: AccountId = receiver_id.into();
+
+        // Get the unused amount from the `ft_on_transfer` call result.
+        let unused_amount = match env::promise_result(0) {
+            PromiseResult::NotReady => unreachable!(),
+            PromiseResult::Successful(value) => {
+                if let Ok(unused_amount) = near_sdk::serde_json::from_slice::<U128>(&value) {
+                    std::cmp::min(amount, unused_amount.0)
+                } else {
+                    amount
+                }
+            }
+            PromiseResult::Failed => amount,
+        };
+
+        if unused_amount > 0 {
+            let receiver_balance = self.accounts.get(&receiver_id).unwrap_or(0);
+            if receiver_balance > 0 {
+                let refund_amount = std::cmp::min(receiver_balance, unused_amount);
+                self.accounts.insert(&receiver_id, &(receiver_balance - refund_amount));
+
+                if let Some(sender_balance) = self.accounts.get(&sender_id) {
+                    self.accounts.insert(&sender_id, &(sender_balance + refund_amount));
+                    log!("Refund {} from {} to {}", refund_amount, receiver_id, sender_id);
+                    return (amount - refund_amount).into();
+                } else {
+                    // Sender's account was deleted, so we need to burn tokens.
+                    self.total_supply -= refund_amount;
+                    log!("The account of the sender was deleted");
+                    log!("Burn {}", refund_amount);
+                }
+            }
+        }
+        amount.into()
+    }
+}
+
+impl StorageManager for FungibleToken {
+    fn storage_deposit(&mut self, account_id: Option<ValidAccountId>) -> AccountStorageBalance {
+        let amount = env::attached_deposit();
+        assert_eq!(
+            amount,
+            self.storage_minimum_balance().0,
+            "Requires attached deposit of the exact storage minimum balance"
+        );
+        let account_id =
+            account_id.map(|a| a.into()).unwrap_or_else(|| env::predecessor_account_id());
+        self.internal_register_account(&account_id);
+        AccountStorageBalance { total: amount.into(), available: amount.into() }
+    }
+
+    fn storage_withdraw(&mut self, amount: Option<U128>) -> AccountStorageBalance {
+        assert_one_yocto();
+        let amount: Balance = amount.unwrap_or_else(|| self.storage_minimum_balance()).into();
+        assert_eq!(
+            amount,
+            self.storage_minimum_balance().0,
+            "The withdrawal amount should be the exact storage minimum balance"
+        );
+        let account_id = env::predecessor_account_id();
+        if let Some(balance) = self.accounts.remove(&account_id) {
+            if balance > 0 {
```

Comment:
```suggestion
            if 0 < balance {
```

Reply:
is there some reason for these suggestions?

Reply:
Yes it was linked above.  It's a code style outlined by [here](https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#comparisons), which argues that it's better to match the real number line, e.i.  `x < y` is preferred over `y > x`.

Reply:
I find it hard to read "if zero is less than amount" (0 < amount) while "if amount is greater than zero" sounds organic to me.

@matklad I wonder to hear your input on the rust-analyzer rule.

Reply:
@frol good point.  I'm fine to drop all of the suggestions. Just wanted consistency.

Reply:
I agree with @frol here on the readability when comparing `x` to `0`. I think when you use both variables e.g. `x` and `y` this is when `x < y` is better than `y > x`.

Reply:
`x > 0` is definitely better that `0 < x` most of the time :)

The background behind the rule is that I've found myself doing a lot of bugs in the code involving *multiple* comparisons, like `x < y && x >= z`, or when checking if two intervals overlap. With "always use `<` / `<=`" rule, such comparisons, for me, became less bug prone.
`5 < x && x <= 10` is more intuitive than `x > 5 && x <= 10`. 
`10 < x && x <= 5` is obviously visibly buggy. 

I've clarified the rule to mention *multiple* comparisons in general. 

As usual, common sense should prevail over any written maxims :) Sometimes small wins in readability are outweighed by their discussion.